functions/go/source-gcloud-config/exec/gcloud.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package exec
+
+import (
+	"bytes"
+	"fmt"
+	"os/exec"
+	"regexp"
+	"strings"
+)
+
+var domainRegex = regexp.MustCompile(`\S+@(\S+)`)
+var PullGcloudConfig = PullLocalGcloudConfig
+
+// PullLocalGcloudConfig executes `gcloud` commands to get default gcloud config value.
+// Potential caveat: The gcloud stdout format is based on Google Cloud SDK 351.0.0. If gcloud SDK changes the output
+// format, it might break this generator function on the client-side. Once that happened, this function shall be modified
+// to be able to parse different stdout format according to the installed SDK version.
```

Comment:
@martinmaly Yeah, parsing the config from gcloud file gives a better handling for this caveat. Thank you!